File: api-server/app/models/user_public_datum.rb
class UserPublicDatum < ApplicationRecord
  belongs_to :user

  # excluded_columns = [
  #   'id',
  #   'user_id',
  #   'allow_tips',
  #   'allow_suggested_games',
  #   'timezone',
  #   'profile_about_me',
  #   'profile_age',
  #   'profile_location',
  #   'profile_languages'
  # ]
  #
  # scope :searchable, ->  {
  #   select( UserPublicDatum.attribute_names - excluded_columns )
  # }

  # def select_without columns
  #   select(column_names - columns.map(&:to_s))
  # end

  # Public profile photos and images
  # has_many_attached :profile_images

  validates :user_id, allow_nil: true, presence: true
  validates :username, presence: true
end

File: ember/app/models/user-public-datum.js
import DS from 'ember-data';

export default DS.Model.extend({
  // Belongs to one user
  user: DS.belongsTo('user'),

  // Public profile
  userId: DS.attr('number'),
  username: DS.attr('string'),
  broadcaster: DS.attr('boolean'),
  onlineStatus: DS.attr('boolean'),
  channelTopic: DS.attr('string'),
  currentGameId: DS.attr('number'),
  streamnailPath: DS.attr('string'),
  allowTips: DS.attr('boolean'),
  allowSuggestedGames: DS.attr('boolean'),
  timezone: DS.attr('string'),
  userCustomTags: DS.attr(),
  profilePhotoPath: DS.attr('string'),
  profileGender: DS.attr('string'),
  profileAboutMe: DS.attr('string'),
  profileAge: DS.attr('number'),
  profileLocation: DS.attr('string'),
  profileLanguages: DS.attr('string')
  //profilePlatforms: DS.attr('string')
});
